### PR TITLE
[DOCS] Add throttling based on configuration parameter

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/issues/54862

This PR adds throttling to the transform performance considerations in the transform overview  (https://www.elastic.co/guide/en/elasticsearch/reference/master/transform-overview.html#transform-performance). Since there were similar performance tips in "How transforms work" (https://www.elastic.co/guide/en/elasticsearch/reference/master/transform-checkpoints.html), I've merged the two and linked them.

This PR also edits the API parameter description for docs_per_second, which was added in https://github.com/elastic/elasticsearch/pull/56178. It also sorts the order of the parameters in the create and update transform APIs.

Preview:
* http://elasticsearch_56653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-overview.html#transform-performance
* http://elasticsearch_56653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-checkpoints.html
* http://elasticsearch_56653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/update-transform.html
* http://elasticsearch_56653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-transform.html
* http://elasticsearch_56653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/preview-transform.html

### CODE REVIEW

File: docs/reference/transform/overview.asciidoc [hunk 1]
```diff
@@ -72,4 +72,10 @@ If your {transform} must process a lot of historic data, it has high resource
 usage initially--particularly during the first checkpoint.
 
 For better performance, make sure that your search aggregations and queries are 
-optimized and that your {transform} is processing only necessary data.
+optimized and that your {transform} is processing only necessary data. Consider 
+whether you can apply a source query to the {transform} to reduce the scope of 
+data it processes. Also consider whether the cluster has sufficient resources in 
+place to support both the composite aggregation search and the indexing of its
+results. You can also throttle the rate at which {transforms} perform search and 
+index requests by setting `docs_per_second` when you <<put-transform,create>> or
```
Comment (lcawl) [inline docs/reference/transform/overview.asciidoc:80]: > and index requests

Is this correct that indexing is also affected by this setting?

Comment (hendrikmuhs) [reply docs/reference/transform/overview.asciidoc:80]: yes, by slowing down the searches, you slow down indexing, too. 

Maybe add a sentence that the transform will take longer if throttling is used, although this seems obvious?

and maybe something like:

To know the current `docs_per_second` transform is operating you can take a look into stats (-> link to stats) and calculate the rate with:

```
docs_per_second = documents_processed / search_time_in_ms * 1000
```

(^ I plan to add this to the stats endpoint, however not for 7.8)

File: docs/reference/transform/apis/put-transform.asciidoc [hunk 2]
```diff
@@ -128,6 +128,23 @@ include::{docdir}/rest-api/common-parms.asciidoc[tag=pivot-group-by]
 ====
 //End pivot
 
+//Begin settings
```
Comment (hendrikmuhs) [inline docs/reference/transform/apis/put-transform.asciidoc:131]: :+1:  sorry for the bad placement! I see, it should be ordered, will do better next time!

**Review by hendrikmuhs — APPROVED** [review]
LGTM! added a suggestion.

File: docs/reference/transform/apis/put-transform.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/reference/transform/apis/put-transform.asciidoc:131]: No worries!

File: docs/reference/transform/overview.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/transform/overview.asciidoc]: Thanks, I've added that information in a final paragraph.

**Comment by hendrikmuhs** [comment]
Still LGTM!

**Pull request merged**